src/Data/Either/Open.purs

```diff
@@ -0,0 +1,146 @@
+module Data.Either.Open where
+
+import Data.Either (Either(..))
+import Data.Either.Nested as Nested
+import Data.Void (Void)
+import Partial.Unsafe (unsafePartial)
+
+type Either1 z = Either Void z
+type Either2 a z = Either (Either1 a) z
+type Either3 a b z = Either (Either2 a b) z
+type Either4 a b c z = Either (Either3 a b c) z
+type Either5 a b c d z = Either (Either4 a b c d) z
+type Either6 a b c d e z = Either (Either5 a b c d e) z
+type Either7 a b c d e f z = Either (Either6 a b c d e f) z
+type Either8 a b c d e f g z = Either (Either7 a b c d e f g) z
+type Either9 a b c d e f g h z = Either (Either8 a b c d e f g h) z
+type Either10 a b c d e f g h i z = Either (Either9 a b c d e f g h i) z
+
+in1 :: forall z a. a -> Nested.Either2 z a
+in1 = Right
+
+in2 :: forall z a b. a -> Nested.Either3 z a b
+in2 v = Left (Right v)
+
+in3 :: forall z a b c. a -> Nested.Either4 z a b c
+in3 v = Left (Left (Right v))
+
+in4 :: forall z a b c d. a -> Nested.Either5 z a b c d
+in4 v = Left (Left (Left (Right v)))
+
+in5 :: forall z a b c d e. a -> Nested.Either6 z a b c d e
+in5 v = Left (Left (Left (Left (Right v))))
+
+in6 :: forall z a b c d e f. a -> Nested.Either7 z a b c d e f
+in6 v = Left (Left (Left (Left (Left (Right v)))))
+
+in7 :: forall z a b c d e f g. a -> Nested.Either8 z a b c d e f g
+in7 v = Left (Left (Left (Left (Left (Left (Right v))))))
+
+in8 :: forall z a b c d e f g h. a -> Nested.Either9 z a b c d e f g h
+in8 v = Left (Left (Left (Left (Left (Left (Left (Right v)))))))
+
+in9 :: forall z a b c d e f g h i. a -> Nested.Either10 z a b c d e f g h i
+in9 v = Left (Left (Left (Left (Left (Left (Left (Left (Right v))))))))
+
+in10 :: forall z a b c d e f g h i j. a -> Nested.Either10 (Either z a) b c d e f g h i j
+in10 v = Left (Left (Left (Left (Left (Left (Left (Left (Left (Right v)))))))))
+
+at1 :: forall z r a. r -> (a -> r) -> Nested.Either2 z a -> r
```

Comment:
I opted for supplying a default value instead of just a `Maybe` projection, but I'm happy to do _either_. Nyuk nyuk nyuk.